traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -136,29 +141,160 @@ func UpdateHandler(w http.ResponseWriter, r *http.Request) {
 		return errors.New("either updatePending or revalPending must not be nil")
 	}
-	qry := `UPDATE server SET `
-	updateStrs := []string{}
-	nextI := 1
-	qryVals := []interface{}{}
+
 	if updatePending != nil {
-		updateStrs = append(updateStrs, `upd_pending = $`+strconv.Itoa(nextI))
-		nextI++
-		qryVals = append(qryVals, *updatePending)
+		if *updatePending {
+			if err := dbhelpers.QueueUpdateForServer(tx, serverID); err != nil {
+				return err
+			}
+		} else {
+			if err := dbhelpers.SetApplyUpdateForServer(tx, serverID); err != nil {
+				return err
+			}
+		}
 	}
+
 	if revalPending != nil {
-		updateStrs = append(updateStrs, `reval_pending = $`+strconv.Itoa(nextI))
-		nextI++
-		qryVals = append(qryVals, *revalPending)
+		if *revalPending {
+			if err := dbhelpers.QueueRevalForServer(tx, serverID); err != nil {
+				return err
+			}
+		} else {
+			if err := dbhelpers.SetApplyRevalForServer(tx, serverID); err != nil {
+				return err
+			}
+		}
 	}
-	qry += strings.Join(updateStrs, ", ") + ` WHERE host_name = $` + strconv.Itoa(nextI)
-	qryVals = append(qryVals, hostName)
+	return nil
+}
 
-	if _, err := tx.Exec(qry, qryVals...); err != nil {
-		return errors.New("executing: " + err.Error())
+// UpdateHandler implements an http handler that updates a server's config update and reval times.
+func UpdateHandlerV4(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"id-or-name"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
 	}
-	return nil
+	defer inf.Close()
+
+	idOrName := inf.Params["id-or-name"]
+	serverID, err := strconv.ParseInt(idOrName, 10, 64)
+	if err != nil {
+		id, ok, err := dbhelpers.GetServerIDFromName(idOrName, inf.Tx.Tx)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("getting server id from name '%v': %w", idOrName, err))
+			return
+		} else if !ok {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("server name '"+idOrName+"' not found"), nil)
+			return
+		}
+		serverID = int64(id)
+	}
+
+	cdnName, err := dbhelpers.GetCDNNameFromServerID(inf.Tx.Tx, serverID)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	userErr, sysErr, statusCode := dbhelpers.CheckIfCurrentUserHasCdnLock(inf.Tx.Tx, string(cdnName), inf.User.UserName)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	// Verify query string parameters
+	configUpdateTimeParam, hasConfigUpdateTimeParam := inf.Params["config_update_time"]
+	configApplyTimeParam, hasConfigApplyTimeParam := inf.Params["config_apply_time"]
+	revalidateUpdateTimeParam, hasRevalidateUpdateTimeParam := inf.Params["revalidate_update_time"]
+	revalidateApplyTimeParam, hasRevalidateApplyTimeParam := inf.Params["revalidate_apply_time"]
```

Comment:
I _think_ we landed on `t3c` only sending the _apply_ times but not the _update_ times, right? So this API doesn't really need to allow writing the _update_ times?